src/providers/ad/ad_common.c

```diff
@@ -543,7 +544,7 @@ ad_get_common_options(TALLOC_CTX *mem_ctx,
                       "The hostname [%s] has been expanded to FQDN [%s]. "
                       "If sssd should really use the short hostname, please "
                       "set ad_hostname explicitly.\n", hostname, fqdn);
-                strncpy(hostname, fqdn, sizeof(hostname));
+                strncpy(hostname, fqdn, HOST_NAME_MAX);
```

Comment:
Just for the record I initially thought that it should be (HOST_NAME_MAX - 1) but @alexey-tikhonov corrected me:
https://github.com/SSSD/sssd/blob/035271b72479845f9ef3d2ffe39656cd21926a32/src/providers/ad/ad_common.c#L490